src/base/types/wrappers/SMomentModel.jl

```diff
@@ -1,132 +1,159 @@
+Return a stoichiometry of the [`SMomentModel`](@ref). The enzymatic reactions
+are split into unidirectional forward and reverse ones.
 """
-objective(model::SMomentModel) = model.c
+stoichiometry(model::SMomentModel) =
+    stoichiometry(model.inner) * _smoment_column_reactions(model)
 
 """
-    fluxes(model::SMomentModel)
+    objective(model::SMomentModel)
 
-Returns the reversible reactions in `model`. For 
-the irreversible reactions, use [`reactions`][@ref].
+Reconstruct an objective of the [`SMomentModel`](@ref).
 """
-fluxes(model::SMomentModel) = model.reaction_ids
+objective(model::SMomentModel) = _smoment_column_reactions(model)' * objective(model.inner)
 
 """
-    n_fluxes(model::SMomentModel)
+    reactions(model::SMomentModel)
 
-Returns the number of reversible reactions in the model.
+Returns the internal reactions in a [`SMomentModel`](@ref) (these may be split
+to forward- and reverse-only parts; reactions IDs are mangled accordingly with
+suffixes).
 """
-n_fluxes(model::SMomentModel) = length(model.reaction_ids)
+reactions(model::SMomentModel) =
+    let inner_reactions = reactions(model.inner)
+        [
+            _smoment_reaction_name(inner_reactions[col.reaction_idx], col.direction) for
+            col in model.columns
+        ]
+    end
 
 """
-    irreversible_reactions(model::SMomentModel)
+    n_reactions(model::SMomentModel)
 
-Returns the irreversible reactions in `model`.
+The number of reactions (including split ones) in [`SMomentModel`](@ref).
 """
-reactions(model::SMomentModel) = model.irrev_reaction_ids
+n_reactions(model::SMomentModel) = length(model.columns)
 
 """
-    n_reactions(model::SMomentModel)
+    bounds(model::SMomentModel)
 
-Returns the number of irreversible reactions in `model`.
+Return the variable bounds for [`SMomentModel`](@ref).
 """
-n_reactions(model::SMomentModel) = length(model.irrev_reaction_ids)
+bounds(model::SMomentModel) =
+    ([col.lb for col in model.columns], [col.ub for col in model.columns])
 
 """
-    metabolites(model::SMomentModel)
+    reaction_flux(model::SMomentModel)
 
-Return the metabolites in `model`.
+Get the mapping of the reaction rates in [`SMomentModel`](@ref) to the original
+fluxes in the wrapped model.
 """
-metabolites(model::SMomentModel) = model.metabolites
+reaction_flux(model::SMomentModel) =
+    _smoment_column_reactions(model)' * reaction_flux(model.inner)
 
 """
-    n_metabolites(model::SMomentModel) = 
+    coupling(model::SMomentModel)
 
-Return the number of metabolites in `model`.
+Return the coupling of [`SMomentModel`](@ref). That combines the coupling of
+the wrapped model, coupling for split reactions, and the coupling for the total
+enzyme capacity.
 """
-n_metabolites(model::SMomentModel) = length(metabolites(model))
+coupling(model::SMomentModel) = vcat(
+    coupling(model.inner) * _smoment_column_reactions(model),
+    _smoment_reaction_coupling(model),
+    [col.capacity_required for col in model.columns]',
+)
```

Comment:
Why is this such a large matrix? shouldn't it just be one row for a single enzyme capacity?

Reply:
lines:
- 132 retains the original coupling in the model (discarding that could kill semantics of the original model)
- 133 is for what I misunderstood as arm reactions, except not done the "extra reaction" way but directly. I guess for smoment this can be discarded (see comment above)
- 134 is The One sMoment Row